src/main/java/racing/domain/RacingGame.java

```diff
@@ -0,0 +1,97 @@
+package racing.domain;
+
+import racing.model.Car;
+import racing.model.Racer;
+import racing.model.Racers;
+import racing.strategy.ForwardConditionStrategy;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import static java.util.Arrays.stream;
+import static java.util.stream.Collectors.toCollection;
+import static java.util.stream.IntStream.range;
+
+public class RacingGame {
+    private static final String NEW_LINE = System.lineSeparator();
+
+    private RacingConfig config;
+
+    private int stageCount;
+
+    private RacingGame(ForwardConditionStrategy strategy, String names, int stageCount) {
+        this.config = new RacingConfig(strategy, names, stageCount);
+    }
+
+    public static RacingGame init(ForwardConditionStrategy strategy, String names, int stageCount) {
+        return new RacingGame(strategy, names, stageCount);
+    }
+
+    public String run() {
+        StringBuilder resultDataBuilder = new StringBuilder(NEW_LINE);
+        stageCount = config.stageCount;
+
+        while (isStageGreaterThanZero()) {
+            turnAround(resultDataBuilder);
+            resultDataBuilder.append(NEW_LINE);
+        }
+        return resultDataBuilder
+                .append(config.racers.winner())
+                .toString().trim();
+    }
+
+    private boolean isStageGreaterThanZero() {
+        return 0 < stageCount--;
+    }
+
+    private void turnAround(StringBuilder resultDataBuilder) {
+        config.racers.turnAround(config.strategy, resultDataBuilder);
+    }
+
+    private class RacingConfig {
```

Comment:
```suggestion
    private static class RacingConfig {
```

inner class에 static을 적용할 경우 어떠한 차이가 있을지 공부해보아요

Reply:
중첩 클래스에서 `static`을 사용하지 않으면 외부 인스턴스와 암묵적으로 연결될 수 있고, 메모리 누수등의 사이드 이펙트가 발생할 수 있다고 하네요. 😲